src/classes/CRLP_RollupUI_TEST.cls

```diff
@@ -105,14 +105,63 @@ public class CRLP_RollupUI_TEST {
         System.assertNotEquals(null, jsonModel, 'JSON string of entire model should be returned.');
     }
 
+    /***
+    * @description Verifies if there are no exception thrown in setting up the Rollups when Einstein
+    * insights platform is enabled in the org
+    */
+    @isTest
+    private static void shouldNotThrowExceptionInCRLPSetupWhenEinsteinIsEnabled() {
+        String errMessage = '';
+        String scriptError = 'Script-thrown exception';
+
+        mockRollupCMTValues();
+        Rollup__mdt rollup = CRLP_Rollup_SEL.cachedRollups[0];
+
+        List<String> targetObjectNames = new List<String>{'Account','Contact'};
+        List<String> detailObjectNames = new List<String>{'Opportunity'};
+
+        if (!UTIL_Namespace.isEinsteinInsightsInstalled()) {
+            return;
+        }
+
+        try {
+            String jsonModel = CRLP_RollupUI_SVC.setupRollupDetail(rollup.Id, targetObjectNames, detailObjectNames);
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        System.assert(!errMessage.contains(scriptError));
+        System.assertEquals('', errMessage, 'No exception is returned when Einstein is enabled');
+    }
+
+    /***
+    * @description Validate that the response from getdetailsfieldswithtype does not include the
+    * System field for account object
+    */
+    @isTest
+    private static void shouldNotHaveSystemFieldsInResponseFromGetDetailsType() {
+        List<String> fieldNames = new List<String>();
+
+        mockRollupCMTValues();
+        Rollup__mdt rollup = CRLP_Rollup_SEL.cachedRollups[0];
+
+        String detailObj = 'Account';
+
+        List<Map<String, String>> detailNameToFieldTypeObj = CRLP_RollupUI_SVC.getDetailFieldsWithType(detailObj);
+
+        for(Map<String,String> apiFieldNames: detailNameToFieldTypeObj) {
+            fieldNames.add(apiFieldNames.get('name'));
+        }
+        System.assertEquals(false, fieldNames.contains('npe01__SYSTEM_AccountType__c'), 'Validate that the names does not include the system field');
```

Comment:
I'm not so sure this will work as this is. The "name" property is in the format of `objectName+' '+dfr.name`, plus it's case-sensitive. I recommend the following
```suggestion
        for(Map<String,String> apiFieldNames: detailNameToFieldTypeObj) {
            System.assert(!apiFieldNames.get('name').containsIgnoreCase('npe01__SYSTEM_AccountType__c'),
                'The returned collection of Field API Names should not include the "npe01__SYSTEM_AccountType__c" field');
        }
```

Reply:
I can change it, debug statements with name property came up as api field names. Is it ok to have assert statement inside the for loop?